### PR TITLE
feat: let migration script set jwt expiration meta

### PR DESCRIPTION
**Pull request by artsyjian**
The type of this PR is: **Feat**

This PR solves [PHIRE-1387]

### Description

We have to scan JWTs in Vault for expiry. Plan is to scan for `expires_at` custom metadata. Therefore, migration script should set that meta.

[PHIRE-1387]: https://artsyproduct.atlassian.net/browse/PHIRE-1387?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/migrate_config_secrets/migrate_config_secrets/migrate.py [hunk 1]
```diff
@@ -87,14 +91,38 @@ def migrate_config_secrets(
...
+  ''' extract expiration date from jwt payload and set it in vault custom metadata '''
+  exp_date_filler = 'blah'
+  exp_date_meta_key = 'expires_at'
+  payload = jwt.decode(jwt_str, options={"verify_signature": False, "verify_exp": False})
+  if 'exp' in payload and payload['exp'] is not None:
+    exp_payload = payload['exp']
+    logging.info(f"JWT has exp payload: {exp_payload}")
+    exp_date = datetime.datetime.utcfromtimestamp(exp_payload)
+    logging.info(f"Exp payload to date: {exp_date}")
+    exp_date_str = exp_date.isoformat("T") + "Z"
```
Comment (artsyjian) [inline src/migrate_config_secrets/migrate_config_secrets/migrate.py:123]: this is to agree with the format i see in vault's default time meta, e.g.:
```
ip-192-168-1-200:~ jxu$ vault kv get -mount kvv2 kubernetes/apps/hokusai-sandbox/FOO                                           
================ Secret Path ================                                                                                  
kvv2/data/kubernetes/apps/hokusai-sandbox/FOO                                                                                  
                                                                                                                               
======= Metadata =======
Key                Value
---                -----
created_time       2024-12-03T21:45:18.207664112Z
```

File: src/migrate_config_secrets/migrate_config_secrets/migrate.py [hunk 2]
```diff
@@ -87,14 +91,38 @@ def migrate_config_secrets(
...
+
     # ask user about deleting var in configmap
     answer = input(f'would you like to delete {var} from configmap (y/n)? ')
     if answer == 'y':
       logging.info(f'Deleting {var} from configmap...')
       env_unset(project_repo_dir, artsy_env, var)
+
+def set_exp_meta(vault_client, var, jwt_str):
+  ''' extract expiration date from jwt payload and set it in vault custom metadata '''
+  exp_date_filler = 'nil'
```
Comment (artsyjian) [inline src/migrate_config_secrets/migrate_config_secrets/migrate.py:115]: just suggestion

**Comment by artsyjian** [comment]
omitting tests (for the new lib functions) for now, since we want to get this out.

**Review by mc-jones — APPROVED** [review]
Very nice.

**Pull request merged**